### PR TITLE
List view allows sorting by "Last Modified" and "Name" and "Size"

### PR DESCRIPTION
**Pull request by MasterofJOKers**
This is implemented in JavaScript, because for implementing it in
Python, the concept of creating the directory listing view would have to
be changed.

*Warning: I really don't use JavaScript very often … but it works.

### CODE REVIEW

File: servefile [hunk 1]
```diff
@@ -413,6 +413,54 @@ class DirListingHandler(FileBaseHandler):
...
+        else
+            par.appendChild(a[i+1]);
+    }
+
+    function sort(cls){
+        var arr = Array.prototype.slice.call(document.getElementsByTagName("tr"));
+        var e = arr.shift();
+        if (!e.sort_modifier)
+            e.sort_modifier = 1;
+        e.sort_modifier = -1*e.sort_modifier;
```
Comment (sebageek) [inline servefile]: Could you add whitespace around the `*` and also after the `,` on the next line?

Comment (MasterofJOKers) [reply servefile]: Rebased and force-pushed that with that change.

**Comment by MasterofJOKers** [comment]
* Added sorting by size
* fixed the directory-sorting, which ignored the second-to-last character in comparisons, while it should only have ignored the last (`/`)
* added a comment for better readability
* made the time-sorting, which actually relies on the ordering of the text-represantation, not ignore the last character for directories

**Comment by hartwork** [comment]
Nice! :+1:

**Pull request merged**